sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -251,30 +258,124 @@ export class AssessmentControllerService {
+        identifier,
+        award.valueCode,
+      );
+      finalAward[disbursementValueKey] = null;
+    });
+    // Process the two expected receipts records for federal(FE) and the other for provincial(BC) awards.
     disbursementReceipts
       .filter(
-        (receipt) => receipt.disbursementSchedule.id === disbursementScheduleId,
+        (receipt) =>
+          receipt.disbursementSchedule.id === disbursementSchedule.id,
       )
       .forEach((receipt) => {
-        finalAward[`${identifier}Id`] = receipt.id;
+        // Check if a loan is part of the receipt (e.g. part-time provincial loans shouldn't be available).
+        const loanType = DisbursementReceiptService.getLoanAwardCode(
+          receipt.fundingType,
+          receipt.disbursementSchedule.studentAssessment.offering
+            .offeringIntensity,
+        );
+        if (loanType) {
+          // Add the loan to the list of awards returned.
+          const disbursementLoanKey = `${identifier}${loanType.toLowerCase()}`;
+          finalAward[disbursementLoanKey] = receipt.totalDisbursedAmount;
+        }
+        // Populate the receipt amount in the receipt awards.
+        // If an estimated disbursement award has no equivalent receipt its value will be left as null.
         receipt.disbursementReceiptValues.forEach((receiptValue) => {
-          const disbursementValueKey = `${identifier}${receiptValue.grantType.toLowerCase()}`;
+          if (receiptValue.grantType === BC_TOTAL_GRANT_AWARD_CODE) {
+            // BC Total grants are not part of the students grants and should not be part of the summary.
+            return;
+          }
+          const disbursementValueKey = this.createReceiptFullIdentifier(
+            identifier,
+            receiptValue.grantType,
+          );
           finalAward[disbursementValueKey] = receiptValue.grantAmount;
         });
+        if (receipt.fundingType === RECEIPT_FUNDING_TYPE_PROVINCIAL) {
+          // Create individual BC grants values from BC total grants(BCSG) receipt.
+          // Federal receipts do not contain individual BC grants.
+          const bcTotalGrantAward =
+            disbursementSchedule.disbursementValues.find(
+              (award) => award.valueCode === BC_TOTAL_GRANT_AWARD_CODE,
+            );
+          // If the BC total grants award is not present consider that BC grants were
+          // not eligible and do not need to be returned.
+          if (bcTotalGrantAward) {
+            const bcTotalGrantsReceipt = receipt.disbursementReceiptValues.find(
+              (receipt) => receipt.grantType === BC_TOTAL_GRANT_AWARD_CODE,
+            );
+            // Check if the BC total grants and the receipt have the some total hence.
+            // If the values match the award values from the BC grants can be copied to the summary.
+            const bcGrantsReceiptMatch =
+              bcTotalGrantAward?.valueAmount ===
```

Comment:
the null check for `bcTotalGrantAward` is already performed in condition

Reply:
Changed.